tv_apps/browser/js/toolbar.js

```diff
@@ -1,15 +1,703 @@
+
+    this.searchButton.addEventListener('mouseup',
+        this.clickSearchButton.bind(this));
+
+    this.searchClearButton.addEventListener('mouseup',
+        this.clickSearchClearButton.bind(this));
+
+    this.searchCloseButton.addEventListener('mouseup',
+        this.clickSearchCloseButton.bind(this));
+
+    this.bookmarkButtonBlock.addEventListener('mouseup',
+        Browser.addBookmark.bind(Browser));
+    this.bookmarkButtonAnime.addEventListener('animationend',
+        Browser.bookmarkButtonAnimeEnd.bind(Browser));
+
+    this.showBookmarksButtonBlock.addEventListener('mouseup',
+        Awesomescreen.selectBookmarksTab.bind(Awesomescreen));
+
+    this.homeButtonBlock.addEventListener('mouseup',
+        this.clickHomeButtonBlock.bind(this));
+
+    this.zoomButtonBlock.addEventListener('mouseup',
+        this.clickZoomButtonBlock.bind(this));
+
+    this.tabsButtonBlock.addEventListener('click',
+        Awesomescreen.handleNewTab.bind(Awesomescreen));
+
+    this.newTabButtonBlock.addEventListener('click',
+        Awesomescreen.createAddNewTab.bind(Awesomescreen));
+
+    this.menuButtonBlock.addEventListener('mouseup',
+        this.clickMenuButtonBlock.bind(this));
+    this.historyBlock.addEventListener('mouseup',
+        Awesomescreen.selectHistoryTab.bind(Awesomescreen));
+    this.privateWindowBlock.addEventListener('mouseup',
+        Browser.handlePrivateBrowsing.bind(Browser));
+    this.settingsBlock.addEventListener('mouseup',
+        Settings.show.bind(Settings));
+
+    this.modeButtonBlock.addEventListener('mouseover',
+        this.selectedModeButtonBlock.bind(this));
+    this.modeButtonBlock.addEventListener('mouseout',
+        this.unselectedModeButtonBlock.bind(this));
+
+    this.toolbarPanel.dataset.search = 'false';
+
+    var colorBar = Browser.getColorBarData();
+    this.panCursorButtonBlock.dataset.colorbar = colorBar[0].COLOR;
+    this.zoomButtonBlock.dataset.colorbar = colorBar[1].COLOR;
+    this.tabsButtonBlock.dataset.colorbar = colorBar[2].COLOR;
+    this.newTabButtonBlock.dataset.colorbar = colorBar[3].COLOR;
+
+    // li = 73px * 3(list) + 20px(hover) + 10px(padding)
+    this.menuBlock.style.height = 73 * 3 + 20 + 10 + 'px';
+    Tooltip.init();
+  },
+
+  /**
+   * toolbar bookmark registered check.
+   */
+  isBookmarks: function toolbar_isBookmarks() {
+    if(Toolbar.bookmarkButton.dataset.isbookmark == 'true'){
+      return true;
+    }else{
+      return false;
+    }
+  },
+
+  /**
+   * Set search engine
+   */
+  setSearchEngine: function toolbar_setSearchEngine() {
+    var engineName = SearchUtil.getCurrentEngineName().toLowerCase();
+    // favicon
+    this.searchFavicon.dataset.engine = engineName;
+    // placeholder
+    engineName = engineName.charAt(0).toUpperCase() + engineName.slice(1);
+    this.searchInput.placeholder = engineName;
+    // tooltip
+    this.searchInput.dataset.tips = engineName + ' ' + _('WB_LT_TIPS_SEARCH_BAR');
```

Comment:
This should be:
`navigator.mozL10n.formatValue('WP_LT_TIPS_SEARCH_BAR', {
  engine: engineName
}).then((value) => {
  this.searchINput.dataset.tips = value;
});`

and `WB_LT_TIPS_SEARCH_BAR` should contain engineName as a variable